<a href="https://colab.research.google.com/github/mariferdamian/TC1002/blob/master/world_population.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#María Fernanda Damian
#A0151918
#TC1002S

In [5]:
#Importar librería de Google drive
from google.colab import drive
drive.mount('/content/drive')



Mounted at /content/drive


In [6]:
#Importar librerías 
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [10]:
#Importar Dataset y almacenarlo en una variable
word_population = pd.read_excel('/content/drive/MyDrive/TC1002S/World_population.xlsm')

In [13]:
word_population.head()


Country  ... Total fertility rate
0             Algeria   ...                     
1        ≈land Islands  ...                 1.68
2  Antigua and Barbuda  ...                     
3            Argentina  ...                 2.39
4             Armenia   ...                 1.42

[5 rows x 15 columns]

In [14]:
word_population.shape

(105, 15)

In [30]:
#Crear un análisis descriptivo donde se obtiene la media, desviación estándar, máximos y mínimos, entre otros
word_population.describe(include='all')

Country Continent  ...  Life expectancy at birth female Total fertility rate
count       105       105  ...                       105.000000                  105
unique      105         6  ...                              NaN                   61
top     Bermuda    Europe  ...                              NaN                     
freq          1        35  ...                              NaN                   30
mean        NaN       NaN  ...                        77.394286                  NaN
std         NaN       NaN  ...                         7.305449                  NaN
min         NaN       NaN  ...                        43.100000                  NaN
25%         NaN       NaN  ...                        75.500000                  NaN
50%         NaN       NaN  ...                        78.700000                  NaN
75%         NaN       NaN  ...                        82.700000                  NaN
max         NaN       NaN  ...                        86.400000                  NaN

[11 rows x 15 columns]

Al obtener las estadísitcas descriptivas de cada variable, obtuvimos que el rango de las variables float o enteras son:


*   Date 2007 - 2011 
*   GNI  870 - 86440
*   Life expectancy at birth female 42.2 - 79.9
*   Life expectancy at birth male  43.1 - 86.4

Mientras que la media aritmética y la desviación estandar de cada variable es:
*   Date 
2009.257143   
1.168732
*   GNI  
22902.947368      
20633.995388

*   Life expectancy at birth female 
71.848571          
7.034012
*   Life expectancy at birth male  43.1 - 86.4
77.394286            
7.305449








De estos resultados, podemos inferir que hay una mejor expectativa de vida en los hombres que en la mujeres, ya que es mayor por aproximadamente 6 años.               Asímismo, podemos saber que estos datos son datos del 2007 hasta el 2011.

In [31]:
#Obtener las columnas del dataset
word_population.columns


Index(['Country', 'Continent', 'Date', 'GDP', 'GNI', 'Live births number',
       'Live births rate', 'Deaths number', 'Death rate', 'Rate of increase',
       'Infant deaths number', 'Infant deaths rate (per 1000 births)',
       'Life expectancy at birth male', 'Life expectancy at birth female',
       'Total fertility rate'],
      dtype='object')

In [15]:
#Sacar la mediana de las variables float e intergers
word_population.median()

Date                                2010.0
GNI                                16940.0
Life expectancy at birth male         72.5
Life expectancy at birth female       78.7
dtype: float64

In [33]:
#Ver el margen de error de las variables del dataset
(word_population.isnull().sum()/len(word_population)*100).sort_values(ascending=False)

GNI                                     9.52381
Total fertility rate                    0.00000
Life expectancy at birth female         0.00000
Life expectancy at birth male           0.00000
Infant deaths rate (per 1000 births)    0.00000
Infant deaths number                    0.00000
Rate of increase                        0.00000
Death rate                              0.00000
Deaths number                           0.00000
Live births rate                        0.00000
Live births number                      0.00000
GDP                                     0.00000
Date                                    0.00000
Continent                               0.00000
Country                                 0.00000
dtype: float64

In [35]:
word_population['Date'].min()

2007

In [34]:
word_population['Date'].max()

2011

In [40]:
word_population['Date'].mean()

2009.2571428571428

In [38]:
word_population['GNI'].min()

870.0

In [37]:
word_population['GNI'].max()

86440.0

In [39]:
word_population['GNI'].mean()

22902.947368421053

In [47]:
word_population['Life expectancy at birth female'].min()

43.1